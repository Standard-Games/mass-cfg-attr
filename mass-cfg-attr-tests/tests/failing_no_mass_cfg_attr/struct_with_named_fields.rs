#![deny(unused_must_use, deprecated)]
#![allow(path_statements)] // disable warning for test

#[must_use]
struct ThreeInts {
    #[deprecated]
    first: i16,
    #[deprecated = "testing key-value"]
    second: i8,
    #[deprecated(since = "0.1.0", note = "this is a more complex example")]
    third: i32
}

fn main() {
    // Violates unused_must_use and deprecated without mass_cfg_attr
    ThreeInts {
        first: 1,
        second: 2,
        third: 3
    };
}
