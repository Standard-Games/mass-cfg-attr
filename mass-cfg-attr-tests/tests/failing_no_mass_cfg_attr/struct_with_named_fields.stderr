error: use of deprecated field `ThreeInts::first`
  --> tests/failing_no_mass_cfg_attr/struct_with_named_fields.rs:17:9
   |
17 |         first: 1,
   |         ^^^^^^^^
   |
note: the lint level is defined here
  --> tests/failing_no_mass_cfg_attr/struct_with_named_fields.rs:1:26
   |
1  | #![deny(unused_must_use, deprecated)]
   |                          ^^^^^^^^^^

error: use of deprecated field `ThreeInts::second`: testing key-value
  --> tests/failing_no_mass_cfg_attr/struct_with_named_fields.rs:18:9
   |
18 |         second: 2,
   |         ^^^^^^^^^

error: use of deprecated field `ThreeInts::third`: this is a more complex example
  --> tests/failing_no_mass_cfg_attr/struct_with_named_fields.rs:19:9
   |
19 |         third: 3
   |         ^^^^^^^^

error: unused `ThreeInts` that must be used
  --> tests/failing_no_mass_cfg_attr/struct_with_named_fields.rs:16:5
   |
16 | /     ThreeInts {
17 | |         first: 1,
18 | |         second: 2,
19 | |         third: 3
20 | |     };
   | |______^
   |
note: the lint level is defined here
  --> tests/failing_no_mass_cfg_attr/struct_with_named_fields.rs:1:9
   |
1  | #![deny(unused_must_use, deprecated)]
   |         ^^^^^^^^^^^^^^^
